baseapp/baseapp_test.go

```diff
@@ -204,76 +201,81 @@ func TestLoadVersion(t *testing.T) {
-	app = NewBaseApp(name, logger, db, nil, pruningOpt)
-	app.SetParamStore(&mock.ParamStore{Db: db})
+	app = baseapp.NewBaseApp(name, logger, db, nil, pruningOpt)
 	app.MountStores()
+
 	err = app.LoadLatestVersion()
 	require.Nil(t, err)
+
 	testLoadVersionHelper(t, app, int64(2), commitID2)
 
-	// reload with LoadVersion, see if you can commit the same block and get
-	// the same result
-	app = NewBaseApp(name, logger, db, nil, pruningOpt)
-	app.SetParamStore(&mock.ParamStore{Db: db})
+	// Reload with LoadVersion, see if you can commit the same block and get
+	// the same result.
+	app = baseapp.NewBaseApp(name, logger, db, nil, pruningOpt)
 	err = app.LoadVersion(1)
 	require.Nil(t, err)
+
 	testLoadVersionHelper(t, app, int64(1), commitID1)
+
 	app.BeginBlock(abci.RequestBeginBlock{Header: header})
 	app.Commit()
+
 	testLoadVersionHelper(t, app, int64(2), commitID2)
 }
 
-func useDefaultLoader(app *BaseApp) {
-	app.SetStoreLoader(DefaultStoreLoader)
-}
+func TestSetLoader(t *testing.T) {
+	useDefaultLoader := func(app *baseapp.BaseApp) {
+		app.SetStoreLoader(baseapp.DefaultStoreLoader)
+	}
 
-func initStore(t *testing.T, db dbm.DB, storeKey string, k, v []byte) {
-	rs := rootmulti.NewStore(db, log.NewNopLogger())
-	rs.SetPruning(pruningtypes.NewPruningOptions(pruningtypes.PruningNothing))
-	key := sdk.NewKVStoreKey(storeKey)
-	rs.MountStoreWithDB(key, storetypes.StoreTypeIAVL, nil)
-	err := rs.LoadLatestVersion()
-	require.Nil(t, err)
-	require.Equal(t, int64(0), rs.LastCommitID().Version)
-
-	// write some data in substore
-	kv, _ := rs.GetStore(key).(storetypes.KVStore)
-	require.NotNil(t, kv)
-	kv.Set(k, v)
-	commitID := rs.Commit()
-	require.Equal(t, int64(1), commitID.Version)
-}
+	initStore := func(t *testing.T, db dbm.DB, storeKey string, k, v []byte) {
+		rs := rootmulti.NewStore(db, log.NewNopLogger())
+		rs.SetPruning(pruningtypes.NewPruningOptions(pruningtypes.PruningNothing))
 
-func checkStore(t *testing.T, db dbm.DB, ver int64, storeKey string, k, v []byte) {
-	rs := rootmulti.NewStore(db, log.NewNopLogger())
-	rs.SetPruning(pruningtypes.NewPruningOptions(pruningtypes.PruningDefault))
-	key := sdk.NewKVStoreKey(storeKey)
-	rs.MountStoreWithDB(key, storetypes.StoreTypeIAVL, nil)
-	err := rs.LoadLatestVersion()
-	require.Nil(t, err)
-	require.Equal(t, ver, rs.LastCommitID().Version)
+		key := sdk.NewKVStoreKey(storeKey)
+		rs.MountStoreWithDB(key, storetypes.StoreTypeIAVL, nil)
 
-	// query data in substore
-	kv, _ := rs.GetStore(key).(storetypes.KVStore)
-	require.NotNil(t, kv)
-	require.Equal(t, v, kv.Get(k))
-}
+		require.Nil(t, rs.LoadLatestVersion())
+		require.Equal(t, int64(0), rs.LastCommitID().Version)
 
-// Test that we can make commits and then reload old versions.
-// Test that LoadLatestVersion actually does.
-func TestSetLoader(t *testing.T) {
-	cases := map[string]struct {
-		setLoader    func(*BaseApp)
+		// write some data in substore
+		kv, _ := rs.GetStore(key).(storetypes.KVStore)
```

Comment:
The changes to baseapp tests are essentially captured in this [PR](https://github.com/cosmos/cosmos-sdk/pull/14162). The tests prior to this were messy, scattered all over the place and not well organized.

So in this PR, since I had to port the ABCI 1.0 baseapp tests, I figured I'd clean it up all in one go.